### PR TITLE
Integrate Drone CI into Github Actions Workflow

### PR DESCRIPTION
**Pull request by rayandas**
**What this PR does / why we need it**:
Integrate the Drone CI into upstream's Github Workflow CI so that it is just another step in their CI and everything runs as one. The building/tagging functionality of the CI should remain unchanged, it just needs to be consolidated so we don’t have 2 separate CIs.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes https://github.com/rancher/rancher/issues/35246

**Special notes for your reviewer**:
>  1. I made a separate job `release.yaml` because it should only get triggered when someone tries to tag and release. I didn't add this job inside `ci.yaml` because all the jobs there runs for every pull requests and every push, to prevent running release ci job for every PR, I made it separate. 
> 2. For this commit [remove G109 check till gosec resolves issues](https://github.com/rancher/ingress-nginx/pull/81/commits/38244395a84a102ac86e6b8fd3b8d1d785184922) we had to remove skip one of their rules `G109` as upstream has also done the same. Else security build will fail. more on https://github.com/kubernetes/ingress-nginx/pull/8044

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    container:
+      image: rancher/dapper:v0.5.3
+      env:
+        GIT_IN_DAPPER: true
+        DOCKER_USER: ${{ secrets.docker_username }}
+        DOCKER_PASS: ${{ secrets.docker_password }}
+    steps:
+      - name: Add git
+        run: |
+          apk git add
```
Comment (jiaqiluo) [inline .github/workflows/release.yaml:21]:
```suggestion
          apk add --no-cache git
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -313,3 +313,40 @@ jobs:
...
+    env:
+      PLATFORMS: linux/amd64
+    container:
+      image: rancher/dapper:v0.5.3
+      env:
+        GIT_IN_DAPPER: true
+    steps:
+      - name: Install git
+        run: |
+          apk add git
```
Comment (jiaqiluo) [inline .github/workflows/ci.yaml:345]:
```suggestion
          apk add --no-cache git
```

Also, do you see the job fails if you do not install git here? Is git not builtin?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: tag
+        run: |
+          dapper ci
```
Comment (jiaqiluo) [inline .github/workflows/release.yaml:28]: nit: add a blank line at the end of the file
```suggestion
          dapper ci

```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -313,3 +313,40 @@ jobs:
...
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: build
+        run: |
+          dapper build
```
Comment (jiaqiluo) [inline .github/workflows/ci.yaml:352]: nit: add a blank line at the end of the file
```suggestion
          dapper build

```

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+name: release
+
+on:
+  push:
+    tags:        
+      - '*'
```
Comment (jiaqiluo) [inline .github/workflows/release.yaml:6]: nit: remove trailing spaces 
```suggestion
    tags:
      - '*'
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (rayandas) [reply .github/workflows/ci.yaml:345]: No git is not built-in in both the ubuntu and rancher/dapper images. If git is not installed, the checkout step is downloading the repository using GitHub REST API and while running any git command in the build job, it's failing, and throwing an error saying `fatal: not a git repository (or any of the parent directories): .git`. So before checkout we need to install git. you can refer to the checkout logs here https://github.com/rancher/ingress-nginx/runs/5184047717?check_suite_focus=true

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    container:
+      image: rancher/dapper:v0.5.3
+      env:
+        GIT_IN_DAPPER: true
+        DOCKER_USER: ${{ secrets.docker_username }}
+        DOCKER_PASS: ${{ secrets.docker_password }}
+    steps:
+      - name: Add git
+        run: |
+          apk git --no-cache git
```
Comment (jiaqiluo) [inline .github/workflows/release.yaml:21]:
```suggestion
          apk add --no-cache git
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

**Review by jiaqiluo — APPROVED** [review]

**Pull request merged**